check_json.pl

```diff
@@ -234,26 +277,57 @@
                 $np->add_perfdata(
                     label => lc $label,
                     value => $perf_value,
-                );            
+                );
             }
         }
     }
 }
 
+sub process_wildcard{
+    my($key,$json_response) = @_;
+
+    if ($np->opts->verbose) { (print "DEBUG: handling wildcard on key ". $key. "\n"); }
+    my @parts;
+    my @result;
+    if ($key !~ /\*/) {
+        $Data::Dumper::Terse = 1;
+        $Data::Dumper::Indent = 0;
+        if ($np->opts->verbose) { (print "DEBUG: Now checking: ". $key. " against: ". Dumper($json_response)."\n")};
+        my $output_value = Dumper(eval('$json_response->'.$key));
+        $Data::Dumper::Terse = 0;
+        $Data::Dumper::Indent = 1;
+        return $output_value;
+    } else {
+        @parts = split(/->[\{\[]?\*[\}\]]?->/,$key,2);
+        my $part1 = $parts[0];
+        my $part2 = $parts[1];
+        if ($np->opts->verbose) { (print "DEBUG: processing first part before wildcard: ". $part1 . "\n"); }
+        my $c = eval('$json_response->'.$part1);
+        return if (! $c);
+        if ($np->opts->verbose) { (print "DEBUG: process_wildcard: Fount childs: ".Dumper($c)."\n")};
+        foreach my $v (values($c)){
+            my $r = process_wildcard($part2,$v);
+            push(@result,$r) if $r;
+        }
+    }
+    return @result;
+}
+
 # output some vars in message
 if ($np->opts->outputvars) {
     foreach my $key ($np->opts->outputvars eq '*' ? map { "{$_}"} sort keys %$json_response : split(',', $np->opts->outputvars)) {
         # use last element of key as label
         my $label = (split('->', $key))[-1];
-        # make label ascii compatible
+        # make label ascii compatible i.e. remove the { and }
         $label =~ s/[^a-zA-Z0-9_-]//g;
         my $output_value;
-        $output_value = $json_response->{$key};
+        ## Handle case of wildcard in the middle of the tree: {data}->*->{description}
+        $output_value = join(", ",process_wildcard($key,$json_response));
         push(@statusmsg, "$label: $output_value");
     }
 }
 
 $np->nagios_exit(
     return_code => $result,
-    message     => join(', ', @statusmsg),
+    message     => join('; ', @statusmsg),
```

Comment:
what's the reason for changing this to a semi-colon?

Reply:
better readability.

Reply:
I believe nagios expects this to be separated by commas.  I fear it won't work correctly if that is changed.

Reply:
This is not perfdata, so there is no requirements as far I double checked in the docs. Let me know if you prefer me to revert it.
https://nagios-plugins.org/doc/guidelines.html#PLUGOUTPUT

Reply:
Ah you're right.  Although I think Icinga2 actually relies on it for formatting in the status.  If this breaks any of our users we can revert it.  

👍